Client/Frontend/Browser/BrowserLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,135 @@
+                self.browser.webView.configuration.userContentController.addUserScript(userScript)
+            }
+        }
+    }
+
+    // MARK: - Gesture Recognizer Delegate Methods
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+
+    // MARK: - Long Press Gesture Handling
+    func SELdidLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == UIGestureRecognizerState.Began {
+            //Finding actual touch location in webView
+            var touchLocation = gestureRecognizer.locationInView(self.browser.webView)
+            touchLocation.x -= self.browser.webView.scrollView.contentInset.left
+            touchLocation.y -= self.browser.webView.scrollView.contentInset.top
+            touchLocation.x /= self.browser.webView.scrollView.zoomScale
+            touchLocation.y /= self.browser.webView.scrollView.zoomScale
+
+            evaluateTouchLocation(touchLocation, gestureRecognizer: gestureRecognizer)
+        }
+    }
+
+    /// Check whether long pressed is on valid link. If found valid link then find the rect for link.
+    private func evaluateTouchLocation(touchLocation: CGPoint,gestureRecognizer: UILongPressGestureRecognizer) {
+        self.browser.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler:nil)
+    }
+
+    /// Recursively call block on view and its subviews
+    private func recursiveBlockOnViewAndSubviews(mainView: UIView, block:(view: UIView) -> Void) {
+        block(view: mainView)
+        mainView.subviews.map(){ self.recursiveBlockOnViewAndSubviews($0 as UIView, block) }
+    }
+
+    /// Find location in screen corresponding to webview - in case it is zoomed or scrolled
+    private func rectLocationInWebView(webView:WKWebView,locationRect:CGRect) -> CGRect {
+        var rect = locationRect
+        var scale = self.browser.webView.scrollView.zoomScale
+        rect.origin.x *= scale
+        rect.origin.y *= scale
+        rect.size.width *= scale
+        rect.size.height *= scale
+        rect.origin.x += self.browser.webView.scrollView.contentInset.left;
+        rect.origin.y += self.browser.webView.scrollView.contentInset.top;
+
+        return rect
+    }
+
+    // MARK: - BrowserHelper Mehods
+    class func name() -> String {
+        return "BrowserLongPressGestureRecognizer"
+    }
+
+    func scriptMessageHandlerName() -> String? {
+        return "longPressMessageHandler"
+    }
+
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        var elementsDict: [String:AnyObject]? = message.body as? [String:AnyObject]
+        if elementsDict  == nil {
+            return
+        }
+
+        var elements:[ElementType] = []
```

Comment:
With the dictionary changes mentioned above, this will become <code>var elements = \[LongPressElementType: NSURL\]()</code>, and your append calls below will turn into <code>elements[LongPressElementType.Link] = linkURL</code>.